guppylang/definition/custom.py

```diff
@@ -122,11 +156,9 @@ def _get_signature(self, globals: Globals) -> FunctionType | None:
         )
 
         if requires_type_annotation and not has_type_annotation:
-            raise GuppyError(
-                f"Type signature for function `{self.name}` is required. "
-                "Alternatively, try passing `higher_order_value=False` on definition.",
-                self.defined_at,
-            )
+            err = NoSignatureError(self.defined_at, self.name)
+            err.add_sub_diagnostic(NoSignatureError.Suggestion(None))
```

Comment:
The `None` means that this sub diagnostic does not have span, it's just rendered as text below the main error.

I updated to code so that this subdiagnostic is added by default